src/data/real/ennreal.lean

```diff
@@ -826,6 +826,8 @@ ennreal.sub_eq_of_eq_add hb hc.symm
 @[simp] protected lemma add_sub_cancel_right (hb : b ≠ ∞) : a + b - b = a :=
 (cancel_of_ne hb).add_tsub_cancel_right
 
+lemma sub_self {a : ℝ≥0∞} (h : a ≠ ∞) : a - a = 0 := sub_eq_of_add_eq h $ zero_add a
```

Comment:
Does `tsub_self` prove it for all `a`?

Reply:
Ah of course it does. We can drop this then. Thanks.